Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/Feature/NodeRemoval.php

```diff
@@ -83,6 +88,155 @@ protected function removeRelationRecursivelyFromDatabaseIncludingNonReferencedNo
+                name,
+                position,
+                contentstreamidentifier,
+                dimensionspacepoint,
+                dimensionspacepointhash,
+                parentnodeanchor,
+                childnodeanchor
+            )
+            SELECT name, position, :contentStreamIdentifier AS contentstreamidentifier,
+                dimensionspacepoint, dimensionspacepointhash, parentnodeanchor, childnodeanchor FROM (
+                    /**
+                     * This provides a list of all hierarchy relations to be copied:
+                     * parentnodeanchor and childnodeanchor, name and position only, the rest will be changed
+                     */
+                    WITH RECURSIVE descendantNodes(
+                        relationanchorpoint, parentnodeanchor, name, position, childnodeanchor
+                    ) AS (
+                        /**
+                         * Initial query: find all outgoing tethered child node relations
+                         * from the starting node in its origin
+                         */
+                        SELECT
+                            n.relationanchorpoint,
+                            h.parentnodeanchor,
+                            h.name,
+                            h.position,
+                            h.childnodeanchor
+                        FROM neos_contentgraph_node n
+                             JOIN neos_contentgraph_hierarchyrelation h ON n.relationanchorpoint = h.childnodeanchor
+                        WHERE h.parentnodeanchor = :relationAnchorPoint
+                          AND h.contentstreamidentifier = :contentStreamIdentifier
+                          AND h.dimensionspacepointhash = :originDimensionSpacePointHash
+                          ' . ($event->recursive ? '' : ' AND n.classification = :classification ') . '
+
+                        UNION ALL
+                            /**
+                             * Iteration query: find all outgoing tethered child node relations
+                             * from the parent node in its origin
+                             */
+                            SELECT
+                                c.relationanchorpoint,
+                                h.parentnodeanchor,
+                                h.name,
+                                h.position,
+                                h.childnodeanchor
+                            FROM
+                                descendantNodes p
+                                    JOIN neos_contentgraph_hierarchyrelation h
+                                        ON h.parentnodeanchor = p.relationanchorpoint
+                                    JOIN neos_contentgraph_node c ON c.relationanchorpoint = h.childnodeanchor
+                            WHERE h.contentstreamidentifier = :contentStreamIdentifier
+                              AND h.dimensionspacepointhash = :originDimensionSpacePointHash
```

Comment:
why do we use an originDimensionSpacePointHash inside h.? I don't understand this. Shouldn't we use the affected DSPs here?

-> if this is correct, let's add a documentation why ;)

Reply:
This is correct and already documented above. I'll try to make it more clear with additional comments